### PR TITLE
Bumped API version (TG-8016)

### PR DESCRIPTION
**Pull request by pcrane**
The changes reflect updates to phase building and API bug fixes.

https://diffblue.atlassian.net/browse/TG-8016

### CODE REVIEW

File: src/main/java/com/diffblue/ci/plugin/App.java [hunk 1]
```diff
@@ -219,6 +225,11 @@ public void onReceivedFragment(final Analysis analysis, final Fragment fragment)
     public void onComplete(final Analysis analysis) {
       LOG.info(MessageFormat.format(COMPLETE_FORMAT, analysis.getId()));
     }
+
+    @Override public void onProgress(final Analysis analysis, final Integer integer,
+        final Integer integer1) {
+      // do nothing
```
Comment (pkesseli) [inline src/main/java/com/diffblue/ci/plugin/App.java]: Perhaps log the progress?

File: src/main/java/com/diffblue/ci/plugin/App.java [hunk 2]
```diff
@@ -219,6 +225,11 @@ public void onReceivedFragment(final Analysis analysis, final Fragment fragment)
     public void onComplete(final Analysis analysis) {
       LOG.info(MessageFormat.format(COMPLETE_FORMAT, analysis.getId()));
     }
+
+    @Override public void onProgress(final Analysis analysis, final Integer integer,
```
Comment (pkesseli) [inline src/main/java/com/diffblue/ci/plugin/App.java]: Use more descriptive names than `integer` and `integer1`.

File: src/main/java/com/diffblue/ci/plugin/cover/combine/ClassData.java [hunk 3]
```diff
@@ -303,87 +523,74 @@ public void removeAll(final ClassData that) {
...
-  // https://github.com/ifedorenko/eclipse.jdt.debug/blob/master/org.eclipse.jdt.debug/model/
-  // org/eclipse/jdt/internal/debug/core/breakpoints/JavaBreakpointImportParticipant.java
+  // https://github.com/ifedorenko/eclipse.jdt.debug/blob/master/org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/breakpoints/JavaBreakpointImportParticipant.java
   public static String getMethodSignatureFromNode(final MethodDeclaration node) {
-    @SuppressWarnings("unchecked")
-    final List<SingleVariableDeclaration> params = node.parameters();
+    @SuppressWarnings("unchecked") final List<SingleVariableDeclaration> params = node.parameters();
     final List<String> rparams = getParametersTypeNames(params);
-    if (rparams.size() == params.size())
+    if (rparams.size() == params.size()) {
```
[the remark this answers is not in the document]
Comment (pcrane) [reply src/main/java/com/diffblue/ci/plugin/cover/combine/ClassData.java]: A lot of this class has been copy/pasted from the eclipse plugin (it's annotated with both CoverV2 and Deprecated). I know these changes have been addressed in the eclipse version.

File: src/main/java/com/diffblue/ci/plugin/cover/combine/ClassData.java [hunk 4]
```diff
@@ -303,87 +523,74 @@ public void removeAll(final ClassData that) {
...
   /**
    * Creates a method signature from a specified {@link MethodDeclaration}. The official
    * ]documentation only talks about the method name and the parameters NOT the return type.
    *
-   * @param node The node to create the signature from.
    * @return the signature for the given method node or <code>null</code>
    */
-  // https://github.com/ifedorenko/eclipse.jdt.debug/blob/master/org.eclipse.jdt.debug/model/
-  // org/eclipse/jdt/internal/debug/core/breakpoints/JavaBreakpointImportParticipant.java
+  // https://github.com/ifedorenko/eclipse.jdt.debug/blob/master/org.eclipse.jdt.debug/model/org/eclipse/jdt/internal/debug/core/breakpoints/JavaBreakpointImportParticipant.java
```
[the remark this answers is not in the document]
Comment (pcrane) [reply src/main/java/com/diffblue/ci/plugin/cover/combine/ClassData.java]: A lot of this class has been copy/pasted from the eclipse plugin (it's annotated with both CoverV2 and Deprecated). I know these changes have been addressed in the eclipse version.

File: src/main/java/com/diffblue/ci/plugin/cover/combine/ClassData.java [hunk 5]
```diff
@@ -285,6 +501,10 @@ public void addMethodDeclaration(final MethodDeclaration method) {
     return methods;
   }
 
+  public Set<Annotation> getClassAnnotations() {
```
[the remark this answers is not in the document]
Comment (pcrane) [reply src/main/java/com/diffblue/ci/plugin/cover/combine/ClassData.java]: A lot of this class has been copy/pasted from the eclipse plugin (it's annotated with both CoverV2 and Deprecated). I know these changes have been addressed in the eclipse version.

File: src/main/java/com/diffblue/ci/plugin/App.java [hunk 6]
```diff
@@ -219,6 +234,18 @@ public void onReceivedFragment(final Analysis analysis, final Fragment fragment)
...
     }
+
+    /*
+     * (non-Javadoc)
+     *
+     * @see com.diffblue.cover.lite.IAnalysisListener#onProgress(com.diffblue.cover.lite.Analysis,
+     * Integer, Integer)
+     */
+    @Override
+    public void onProgress(final Analysis analysis, final Integer progress, final Integer total) {
```
[the remark this answers is not in the document]
Comment (pcrane) [reply src/main/java/com/diffblue/ci/plugin/App.java]: I'll address the interface design in 2.0. As this feature isn't provided by the back-end I've removed the implementation here (see: TG-8117).

**Pull request merged**